pkg/config/merge.go

```diff
@@ -0,0 +1,16 @@
+package config
+
+// extendedService is merged into service
+func merge(service *Service, extendedService *Service) {
+	// rules based on https://docs.docker.com/compose/extends/#adding-and-overriding-configuration
+	mergeMap(service.Environment, extendedService.Environment)
+	if len(extendedService.Image) > 0 {
+		service.Image = extendedService.Image
```

Comment:
Image is only extended in docker-compose V1, apparently: https://github.com/docker/compose/blob/master/compose/config/config.py#L1128

Reply:
will remove this for now